app/models/concerns/obstructed.rb

```diff
@@ -0,0 +1,70 @@
+# In queen.rb, rook.rb, bishop.rb, pawn.rb, I added "include Obstructed"
+# Every time one of these pieces is moved the "valid_move?" method should be called
+# and that should call the "is_obstructed?" method.
+# game.pieces.where(horizontal_position & vertical_position) are extracted current location of pieces on game board.
+# If is_obstructed returns false, move is legal.
+# x_start, x_end, y_start, y_end values are filled when piece is moved.
+# Will write tests later after moving pieces action is more clear...
+module Obstructed
+  def horizontal_check
+    if x_end < x_start
+      (x_end + 1..x_start - 1).each do |x_between|
+        return true if game.pieces.where(horizontal_position: x_between).present?
+      end
+    else
+      (x_start + 1..x_end - 1).each do |x_between|
+        return true if game.pieces.where(horizontal_position: x_between).present?
+      end
+    end
+    false
+  end
+
+  def vertical_check
+    if y_end < y_start
+      (y_end + 1..y_start - 1).each do |y_between|
+        return true if game.pieces.where(vertical_position: y_between).present?
+      end
+    else
+      (y_start + 1..y_end - 1).each do |y_between|
+        return true if game.pieces.where(vertical_position: y_between).present?
+      end
+    end
+    false
+  end
+
+  def diagonal_check

```

Comment:
I think the complexity warnings from Rubocop are because there is repeated code that can be refactored. The block from line 38 to 45 and the block from 47 to 54 are almost identical code except for some of the x/y-start vars and whether you add or subtract counts. So you could refactor it to reduce some of the duplication. One way to start refactoring could be to instead of having branches for both going left or right, just always go right (so your start location would be = ((x_start < x_end) ? x_start : x_end)),  and the end condition would be the other). 

Another approach could be to have a helper method for all the check methods that builds an array of all the coordinates between the start location and the end location and then just loop through that array of coordinates.